### PR TITLE
Remove test dependencies from the dependencies list

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Review by bowenszhu — COMMENTED** [review]
Better to follow the new test environment format?
https://pkgdocs.julialang.org/v1/creating-packages/#Test-specific-dependencies-in-Julia-1.2-and-above

File: Project.toml [hunk 1]
```diff
@@ -12,16 +12,13 @@ Interpolations = "a98d9a8b-a2ab-59e6-89dd-64a1c18fca59"
 Latexify = "23fbe1c1-3f47-55db-b15f-69d7ec21a316"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 ModelingToolkit = "961ee093-0014-501f-94e3-6117800e7a78"
-NonlinearSolve = "8913a72c-1f9b-4ce2-8d82-65094dcecaec"
 OrdinaryDiffEq = "1dea7af3-3e70-54e6-95c3-0bf5283fa5ed"
 SafeTestsets = "1bc83da4-3b8d-516f-aca4-4fe02f6d838f"
```
Comment (bowenszhu) [inline Project.toml:16]: And SafeTestsets should also be removed.

Comment (ChrisRackauckas) [reply Project.toml]: Yep definitely thanks.

**Comment by ChrisRackauckas** [comment]
Probably, but I'm behind on a few things and this is quicker so it's at least a fix for now. Getting the new NonlinearSolve out there and getting the enum handling fixes all done ASAP so I can get back to other things is my main priority, and then @xtalax can handle whatever else from there.

**Comment by xtalax** [comment]
Sure, I've seen this now - I had thought that there should be a way to have testing dependencies separated but didn't know how to go about doing it, so thanks.

What actions do I need to take after this is merged? Follow the guide?

**Pull request merged**